packages/web/hooks/use-has-asset-variants.tsx

```diff
@@ -0,0 +1,57 @@
+import { useLocalStorage } from "react-use";
+
+import {
+  AlloyedAssetsToastDoNotShowKey,
+  displayToast,
+} from "~/components/alert/toast";
+import { ToastType } from "~/components/alert/types";
+import { useFeatureFlags, useWindowSize } from "~/hooks";
+import { useStore } from "~/stores";
+import { api } from "~/utils/trpc";
+
+export const useHasAssetVariants = () => {
+  const { accountStore } = useStore();
+  const wallet = accountStore.getWallet(accountStore.osmosisChainId);
+
+  const { isMobile } = useWindowSize();
+  const { alloyedAssets } = useFeatureFlags();
+
+  const [doNotShowAgain] = useLocalStorage(
+    AlloyedAssetsToastDoNotShowKey,
+    false
+  );
+
+  const enabled =
+    alloyedAssets &&
+    !doNotShowAgain &&
+    Boolean(wallet?.isWalletConnected && wallet?.address);
+
+  api.local.portfolio.getHasAssetVariants.useQuery(
```

Comment:
From looking at implementation of `useHasAssetVariants`

What if we just run that function inside `getAllocation` since we operate on that same data that we query elsewhere anyways? Then `useHasAssetVariants` here would call `getAllocation` looking for that `hasAssetVariants: boolean` flag instead when deciding to show the toast.

Pros:
* Can remove a redundant tRPC procedure call (this one, `getHasAssetVariants`) in `portfolioRouter`
* Prevent redundant API call to SQS for `passthrough/portfolio-assets`, reducing load on SQS services and avoiding an unnecessary query from the browser
* If we check for convertible assets on every page anyways, we can reuse that allocation query data when they navigate to the portfolio page, increasing navigation performance

Cons:
* May be hard to know that flag is there. In other words, the purpose of `getAllocation` has expanded a bit to involve alloyed assets logic

Overall, I think it's worth it, as we can delete a bunch of code. Which means less chance for error, and less to test.

Reply:
I'm seeing redundant calls to portfolio-assets from different initiator call stacks.

I think it could easily be avoided, and could save on bandwidth to SQS:

<img width="414" alt="Screenshot 2024-10-07 at 6 03 02 PM" src="https://github.com/user-
attachments/assets/bbaf8c65-128e-4037-80e6-219bf1830d83">

Reply:
@mattupham 

Check this out
https://github.com/osmosis-labs/osmosis-frontend/pull/3872